xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -285,6 +285,41 @@ bool CGUIWindowPVRBase::OnContextButtonActiveAEDSPSettings(CFileItem *item, CONT
   return bReturn;
 }
 
+bool CGUIWindowPVRBase::OnContextButtonEditTimer(CFileItem *item, CONTEXT_BUTTON button)
+{
+  bool bReturn = false;
+
+  if (button == CONTEXT_BUTTON_EDIT_TIMER)
+  {
+    EditTimer(item);
+    bReturn = true;
+  }
+
+  return bReturn;
+}
+
+bool CGUIWindowPVRBase::OnContextButtonEditTimerRule(CFileItem *item, CONTEXT_BUTTON button)
+{
+  bool bReturn = false;
+
+  if (button == CONTEXT_BUTTON_EDIT_TIMER_RULE)
+  {
+    if (item && item->IsEPG())
+    {
+      const CPVRTimerInfoTagPtr timer(item->GetEPGInfoTag()->Timer());
+      if (timer)
+      {
+        const CPVRTimerInfoTagPtr timerRule(g_PVRTimers->GetTimerRule(timer));
+        if (timerRule)
+          EditTimer(CFileItemPtr(new CFileItem(timerRule)).get());
```

Comment:
Could EditTimer take a timer info tag pointer instead of a file item? It would reduce the need for arbitrarily wrapping stuff in file items.